File: mpv-util.py
import sys
import json

def main():
    # Ensure there is an action argument passed
    if len(sys.argv) < 2:
        return

    action = sys.argv[1]
    
    # Read all inputs coming from the Bash redirection (<<<) via stdin
    try:
        input_data = sys.stdin.read().strip()
    except Exception:
        input_data = ""

    if action == "--get-prop":
        # Wraps a raw property name string into an mpv JSON IPC command payload
        print(json.dumps({"command": ["get_property", input_data]}))

    elif action == "--parse-prop":
        # Parses mpv socket response and extracts the underlying 'data' field safely
        try:
            obj = json.loads(input_data)
            if "data" in obj and obj["data"] is not None:
                print(obj["data"])
        except Exception:
            pass

    elif action == "--send-cmd":
        # Parses raw string array arguments and embeds them inside an execution command
        try:
            args = json.loads(input_data)
            print(json.dumps({"command": args}))
        except Exception:
            pass

    elif action == "--cache-bytes":
        # Drills deep into the demuxer cache object structure to isolate total-bytes
        try:
            obj = json.loads(input_data)
            data = obj.get("data")
            if data and "total-bytes" in data:
                print(data["total-bytes"])
            else:
                print(0)
        except Exception:
            print(0)

if __name__ == "__main__":
    main()

File: mpv-util.js
const fs = require('fs');

// Read all inputs coming from the Bash redirection (<<<) via stdin
const input = fs.readFileSync(0, 'utf-8').trim();
const action = process.argv[2];

switch (action) {
    case '--get-prop':
        // Wraps a raw property name string into an mpv JSON IPC command payload
        console.log(JSON.stringify({ "command": ["get_property", input] }));
        break;

    case '--parse-prop':
        // Parses mpv socket response and extracts the underlying 'data' field safely
        try {
            const obj = JSON.parse(input);
            if (obj.data !== undefined && obj.data !== null) {
                console.log(obj.data);
            }
        } catch (e) {}
        break;

    case '--send-cmd':
        // Parses raw string array arguments and embeds them inside an execution command
        try {
            const args = JSON.parse(input);
            console.log(JSON.stringify({ "command": args }));
        } catch (e) {}
        break;

    case '--cache-bytes':
        // Drills deep into the demuxer cache object structure to isolate total-bytes
        try {
            const obj = JSON.parse(input);
            if (obj.data && obj.data["total-bytes"]) {
                console.log(obj.data["total-bytes"]);
            } else {
                console.log(0);
            }
        } catch (e) {
            console.log(0);
        }
        break;
}
